### PR TITLE
Fix inconsistent cluster state and ensure weigh away exception check only for data nodes

### PR DESCRIPTION
**Pull request by anshu1106**
### Description
This PR fixes inconsistent cluster state with ensure_weighed_in param while make cluster health api call.
Also adds support for weighed away health check only for data nodes.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java [hunk 1]
```diff
@@ -280,15 +280,16 @@ private void executeHealth(
...
             ClusterHealthResponse clusterHealthResponse = getResponse(request, currentState, waitCount, TimeoutState.OK);
             if (request.ensureNodeWeighedIn() && clusterHealthResponse.hasDiscoveredClusterManager()) {
-                DiscoveryNode localNode = clusterService.state().getNodes().getLocalNode();
-                assert request.local() == true : "local node request false for request for local node weighed in";
-                boolean weighedAway = WeightedRoutingUtils.isWeighedAway(localNode.getId(), clusterService.state());
-                if (weighedAway) {
-                    listener.onFailure(new NodeWeighedAwayException("local node is weighed away"));
-                    return;
+                DiscoveryNode localNode = currentState.getNodes().getLocalNode();
+                if (localNode.isDataNode()) {
```
Comment (Bukhtawar) [inline server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java:284]: Should it be made more generic based on roles?

Comment (anshu1106) [reply server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java:284]: we can do that but can we say that all data nodes have `DATA_ROLE`? And ultrawarm nodes don't have `DATA_ROLE`

Comment (anshu1106) [reply server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java:285]: added a `todo` to take this later

File: CHANGELOG.md [hunk 2]
```diff
@@ -122,6 +122,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [Segment Replication] Fix for peer recovery ([#5344](https://github.com/opensearch-project/OpenSearch/pull/5344))
 - Fix weighted shard routing state across search requests([#6004](https://github.com/opensearch-project/OpenSearch/pull/6004))
 - [Segment Replication] Fix bug where inaccurate sequence numbers are sent during replication ([#6122](https://github.com/opensearch-project/OpenSearch/pull/6122))
+- [Weighted Routing] Fix inconsistent cluster state and ensure weigh away exception check only for data nodes ([#6327](https://github.com/opensearch-project/OpenSearch/pull/6327))
```
Comment (gbbafna) [inline CHANGELOG.md:125]: lets skip this as it is not a user facing change.

**Review by gbbafna — APPROVED** [review]

**Review by Bukhtawar — APPROVED** [review]

**Review by Bukhtawar — APPROVED** [review]

**Pull request merged**